gtsfm/bundle/bundle_adjustment.py

```diff
@@ -122,24 +124,31 @@ def run(self, initial_data: GtsfmData) -> GtsfmData:
         # construct the results
         optimized_data = values_to_gtsfm_data(result_values, initial_data)
 
-        metrics_dict = {}
-        metrics_dict["before_filtering"] = optimized_data.aggregate_metrics()
-        logger.info("[Result] Number of tracks before filtering: %d", metrics_dict["before_filtering"]["number_tracks"])
+        def get_metrics_from_sfm_data(sfm_data: GtsfmData, suffix: str) -> List[GtsfmMetric]:
+            """Helper to get bundle adjustment metrics from a GtsfmData object with a suffix for metric names."""
+            metrics = []
+            metrics.append(GtsfmMetric("number_tracks" + suffix, sfm_data.number_tracks()))
+            metrics.append(GtsfmMetric("3d_track_lengths" + suffix, sfm_data.get_track_lengths()))
+            metrics.append(GtsfmMetric("reprojection_errors" + suffix, sfm_data.get_scene_reprojection_errors()))
+            return metrics
```

Comment:
can this be a function of the `GtsfmData` class. I think your recent PR about having just core logic in the modules is a good idea and we should follow it here.

Reply:
It makes sense to have them as a method of `GtsfmData`. But then it would be more like "track metrics" rather than bundle adjustment metrics. 

So a method like `def get_track_metrics(suffix="") -> List[GtsfmMetric]:` could be added. If that sounds good to you, I can add it in this PR, else feel free to  create another quick PR for that once this is merged.

Reply:
let's do in the next PR